test/celeritas/field/FieldPropagator.test.cc

```diff
@@ -887,6 +903,112 @@ TEST_F(TwoBoxTest, electron_step_endpoint)
     }
 }
 
+// Electron barely crosses boundary
+TEST_F(TwoBoxTest, electron_tangent_cross_smallradius)
+{
+    auto particle = this->init_particle(
+        this->particle()->find(pdg::electron()), MevEnergy{10});
+
+    UniformZField   field(unit_radius_field_strength * 100);
+    const real_type radius        = 0.01;
+    const real_type miss_distance = 1e-4;
+
+    std::vector<int>       boundary;
+    std::vector<real_type> distances;
+    std::vector<int>       substeps;
+
+    for (real_type dtheta : {pi / 4, pi / 7, 1e-3, 1e-6, 1e-9})
+    {
+        SCOPED_TRACE(dtheta);
+        {
+            // Angle of intercept with boundary
+            real_type tint = std::asin((radius - miss_distance) / radius);
+            const real_type sintheta = std::sin(tint - dtheta);
+            const real_type costheta = std::cos(tint - dtheta);
+
+            Real3 pos{radius * costheta,
+                      5 + miss_distance - radius + radius * sintheta,
+                      0};
+            Real3 dir{-sintheta, costheta, 0};
+            cout << "pos=" << pos << ", dir=" << dir << endl;
+            this->init_geo(pos, dir);
+        }
+        auto geo = this->make_geo_view();
+
+        EXPECT_SOFT_EQ(radius,
+                       this->calc_field_curvature(particle, geo, field));
+
+        // Build propagator
+        auto stepper = make_mag_field_stepper<DiagnosticDPStepper>(
+            field, particle.charge());
+        FieldDriverOptions driver_options;
+        auto               propagate
+            = make_field_propagator(stepper, driver_options, particle, &geo);
+        for (int i : range(2))
+        {
+            SCOPED_TRACE(i);
+            auto result = propagate(radius * dtheta);
+
+            boundary.push_back(result.boundary);
+            distances.push_back(result.distance);
+            substeps.push_back(stepper.count());
+            stepper.reset_count();
+            if (!CELERITAS_USE_VECGEOM)
+            {
+                // Correct
+                EXPECT_EQ(std::string(i == 0 ? "inner" : "outer"),
+                          this->volume_name(geo));
+            }
+            else
+            {
+                // INCORRECT: vecgeom doesn't return zero/1e-8 distance
+                EXPECT_EQ("inner", this->volume_name(geo));
+            }
```

Comment:
So, why is this expect result with vecgeom different from orange?  should the expect volume "inner" for both? (so, are both incorrect or correct as moving by step/2 now?)

Reply:
No, the expected should be alternating inner/outer (since the first segment is to the boundary on the inside, the second segment is the tiny dip in the outer volume). I just made a stupid mistake and forgot to cross the boundary: now both geometries are giving the same answer. Thanks!